pkg/arvo/lib/lv.hoon

```diff
@@ -0,0 +1,451 @@
+  ++  divv  (funv div:rs)
+  ::
+  ::    Sum of elements
+  ++  sum
+    |=  [u=@lvs]  ^-  @rs
+    (roll (unmake u) add:rs)
+  ::
+  ::    Cumulative sum of elements
+  ++  cumsum
+    |=  [u=@lvs]  ^-  @lvs
+    =/  n  (length u)
+    =/  uu  (unmake u)
+    =/  v  (zeros n)
+    =/  index  1
+    |-  ^-  @lvs
+      ?:  (gth index n)  v
+    $(index +(index), v (set v index (sum (subvector u 1 index))))
+  ::
+  ::    Product of elements
+  ++  product
+    |=  [u=@lvs]  ^-  @rs
+    (roll (unmake u) |:([a=.1 b=.1] (mul:rs a b)))
+  ::
+  ::  Linear algebraic operators
+  ::
+  ::    Inner or Euclidean dot product, a · b
+  ++  inner
+    ::~/  %inner
+    |=  [u=@lvs v=@lvs]  ^-  @rs
+    ~_  leaf+"lagoon-fail"
+    (sum (mulv u v))
+  ++  outer  !!  :: unimplemented pending @lm type
+  ++  catenate
+    |=  [u=@lvs v=@lvs]
+    ~_  leaf+"lagoon-fail"
+    (make (weld (unmake u) (unmake v)))
+    :: XX slow way, do in bits
+  --
+::
+::::  Vector type in double-precision floating-point @rd
+  ::
+++  lvd
+  ^|
+  ::~%  %lvd  +>  ~
+  |_  r=$?(%n %u %d %z)   :: round nearest, round up, round down, round to zero
+  ::
+  ::  Manipulators
+  ::
+  ::    Zeroes
+  ++  zeros
+    ::~/  %zeros
+    |=  n=@ud  ^-  @lvd
+    ~_  leaf+"lagoon-fail"
+    `@lvd`(lsh [6 n] 1)   :: pin at head for leading zeros
+  ::
+  ::    Fill value
+  ++  fill
+    |=  [n=@ud s=@rd]  ^-  @lvd
+    `@lvd`(mix (zeros n) (fil 6 n s))
+  ::
+  ::    Ones
+  ++  ones
+    |=  n=@ud  ^-  @lvd
+    ~_  leaf+"lagoon-fail"
+    (fill n .~1)
+  ::
+  ::    Length of vector
+  ++  length
+    |=  u=@lvd  ^-  @ud
+    ~_  leaf+"lagoon-fail"
+    =/  ell  (met 6 u)
+    ?:  (gth ell 1)  (dec ell)  0
+  ::
+  ::    Produce a vector from `(list @u)` (of natural numbers)
+  ++  make-nat
+    |=  a=(list @u)  ^-  @lvd
+    (make (turn a sun:rd))
+  ::
+  ::    APL-style index list
+  ++  iota
+    |=  n=@u  ^-  @lvd
+    (make-nat (gulf 1 n))
+  ++  make
+    |=  [a=(list @rd)]  ^-  @lvd
+    ~_  leaf+"lagoon-fail"
+    `@lvd`(mix (rep [6 1] a) (zeros (lent a)))
+  ++  unmake
+    |=  [u=@lvd]  ^-  (list @rd)
+    ~_  leaf+"lagoon-fail"
+    (flop `(list @rd)`+:(flop (rip 6 u)))
+  ++  append
+    |=  [u=@lvd s=@rd]  ^-  @lvd
+    (make (snoc (unmake u) s))
+    ::  XX could be done faster with a mix/lsh
+  ::
+  ::  Yield the substring [lhs:rhs] inclusive
+  ++  subvector
+    |=  [u=@lvd lhs=@ud rhs=@ud]
+    (mix (zeros +((sub rhs lhs))) (cut 6 [(dec lhs) +((sub rhs lhs))] u))
+  ::
+  ::  |x|
+  ++  abs
+    |=  [s=@rd]
+    ?:  (gth:rd s .~0)  s  (sub:rd .~0 s)
+  ::
+  ::  |x-y| <= tol
+  ++  isclose
+    |=  [s=@rd t=@rd tol=@rd]
+    (lth:rd (abs (sub:rd s t)) tol)
+  ++  near0
+    |=  s=@rd
+    (isclose s .~0 .~1e-6)
+  ++  all-close
+    |=  [u=@lvd v=@lvd]
+      =/  n  `@ud`(length u)
+      =/  count  1
+      =/  off  0
+      |-
+        ?:  (gth count n)  =(0 off)
+      $(count +(count), sum (add off `@`(isclose (get u +(count)) (get v +(count)) .~1e-6)))
```

Comment:
The test failure is pointing to this line: you're replacing `sum` and recursing, but that's not a local face (and is presumably a core in the stdlib). Is it supposed to be `off (add off ...)`?